src/user-event/type/type.ts

```diff
@@ -94,6 +95,7 @@ export async function emitTypingEvents(
     return;
   }
 
+  nativeState?.elementValues.set(element, text);
```

Comment:
i'm curious, why do you set the native state on each key press and not at the end (after the `  for (const key of keys)`, if i understand correctly, for each key, of the text, we dispatch an event and we set the initial value right ?

Reply:
If we make the native state accessible through refs (not there yet, but could be). Then the event handler could theoretically query the new state. Otherwise, it's practially zero cost so why not.